test/extended/baremetal/high_availability.go

```diff
@@ -0,0 +1,117 @@
+package baremetal
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	interval           = 10 * time.Second
+	waitTimeout        = 5 * time.Minute
+	baremetalNamespace = "openshift-machine-api"
+)
+
+var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platform should ensure", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("baremetal")
+
+	g.It("cluster baremetal operator returns back healthy after it is deleted", func() {
+		skipIfNotBaremetal(oc)
+
+		c, err := e2e.LoadClientset()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("delete cluster baremetal operator")
+		err = c.AppsV1().Deployments(baremetalNamespace).Delete(context.Background(), "cluster-baremetal-operator", v1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("wait until cluster baremetal operator returns back healthy")
+		err = wait.Poll(interval, waitTimeout, func() (bool, error) {
+			dc, err := c.AppsV1().Deployments(baremetalNamespace).Get(context.Background(), "cluster-baremetal-operator", v1.GetOptions{})
+			if err != nil {
+				// This error is expected at first run
+				if strings.Contains(err.Error(), `deployments.apps "cluster-baremetal-operator" not found`) {
+					return false, nil
+				}
+				return false, err
+			}
+
+			if dc.Status.AvailableReplicas != 1 {
+				return false, nil
+			}
+
+			return true, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("verify that baremetal hosts are healthy and correct state")
+		checkMetal3DeploymentHealthy(oc)
```

Comment:
I'm not sure about the usefulness of this check in case of a CBO delete

Reply:
I don't agree. A couple of months ago, there was a bug that caused nodes are stuck in registration error because of a bug in BMO or CBO. I think, it is useful to check that this deletion(I mean both cases) must not have any effect on the bmh statuses. I think, this is even more important than CBO permanently stays in degraded status.

Reply:
It seems different things get mixed, and maybe a different approach could be required. The goal of this test is to verify that CBO returns operative after a deletion. Since the main task of CBO is to deploy a metal3 deployment when missing, one possible way to check that CBO is continuing to operate as normal would be to delete the metal3 deployment _after_ CBO came back, and then wait for it to become ready (and this probably would mean merging the case below here)

Reply:
Why do you think different things are mixed because by deleting CBO and only waiting it returns back is testing the CVO nothing more than that. For all baremetal tests, I think we always ensure that BMHs are in correct state(all operators we have exist for baremetalhosts).

But I agree with you, we can boil these two tests down to one test(deleting sequentially) seems like a more correct approach. Delete CBO, wait ready, delete metal3, wait ready, and check bmh statuses?

Reply:
Mainly due the check on the pre-existing resources. Thus the `Delete CBO, wait ready, delete metal3, wait ready, and check bmh statuses` sequence seems fine to me for such test :+1:

Reply:
Thanks @andfasano, this test will be a good candidate catching possible errors.